tools/ReleaseNotes/release_editor.py

```diff
@@ -0,0 +1,129 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import os
+import sys
+import shutil
+
+DIRECTIVE = ".. amalgamate:: "
+
+
+def getReleaseRoot():
```

Comment:
I know the syle here roughly follows the `release.py` script but that is a little old now. Could we modernize a bit and use [pathlib](https://realpython.com/python-pathlib/#creating-paths) when dealing with paths? For example,

```python
def getReleaseRoot() -> pathlib.Path:
    program_path = pathlib.Path(sys.argv[0])
    return program_path.resolve('../../docs/source/release')
```

This will mean we stop treating paths as strings and allows Python internally to worry about the differences betwen `\` and `/` on Windows. This function now returns a `Path` and not a `str` so I've snuck in a `-> pathlib.Path` on the end of the function to indicate this. This is, [type annotation](https://realpython.com/python-type-checking/#annotations) a good habit for more Python too as it makes it easier to read what types. are expected

Reply:
Moving to using pathlib is a good idea. However it has required quite a bit of re-writing to make the whole script work. That being said I think it looks much better now.

For this one I don't think .resolve() takes any arguments so I had to include an extra step to get us where we needed.